sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JsonSerializer.java

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.serializer;
+
+import java.io.OutputStream;
+
+/**
+ * Generic interface covering basic JSON serialization and deserialization methods.
+ */
+public interface JsonSerializer {
+    /**
+     * Reads the JSON byte stream into its object representation.
+     *
+     * @param input JSON byte stream.
+     * @param clazz {@link Class} representing the object.
+     * @param <T> Type of the object.
+     * @return The object representing the JSON string.
+     */
+    <T> T deserialize(byte[] input, Class<T> clazz);
+
+    /**
+     * Writes the object into its JSON byte stream.
+     *
+     * @param value The object.
+     * @return The JSON byte stream representing the object.
+     */
+    byte[] serialize(Object value);
+
+    /**
+     * Converts the object into a JSON byte stream and writes it to the {@link OutputStream}.
+     *
+     * @param value The object.
+     * @param stream The {@link OutputStream} where the JSON byte stream will be written.
+     */
+    void serialize(Object value, OutputStream stream);
```

Comment:
Is there an immediate use case for this API?

Reply:
This should be removable and has a viable work around without the existence of the API.

**With API**

```java
serialize(value, stream);
```

**Without API**

```java
stream.write(serialize(value));
```

Reply:
Is there a use case though that the API would enable, in particular with greater perf?

Reply:
There may be a minor performance enhancement but the underlying serialization logic is generally the same when using either `write(Object)` or `write(Object, OutputStream)` APIs. Given the API is for a pattern we generally don't see in our libraries I don't think keeping it adds much worth.